api.js

```diff
@@ -17,24 +18,25 @@ function init(page, cb, prevSpeed) {
 	})
 	.then(result => {
 		if (result.speed > 0 && result.speed !== prevSpeed) {
-			cb(null, result);
+			observer.next(result);
 		}
 
 		if (result.isDone) {
 			page.close();
+			observer.complete();
 		} else {
-			setTimeout(init, 100, page, cb, result.speed);
+			setTimeout(init, 100, page, observer, result.speed);
 		}
 	})
-	.catch(cb);
+	.catch(err => observer.error(err));
```

Comment:
I think the `error` method needs the context of `observer`, would `observer.error.bind(observer)` be preferable?